### ISSUE TITLE
Invalid (?) "Insufficient access rights" on itemDependencyCreate

### ISSUE THREAD

**Issue by smadbe** [France-ioi/AlgoreaBackend#851]
```sh
curl 'https://dev.algorea.org/api/items/128139237432513103/prerequisites/902356740789159624' -X POST -H 'Authorization: Bearer <usual_token>' --data-raw '{"grant_content_view":true}'
{"success":false,"message":"Forbidden","error_text":"Insufficient access rights"}
```
while the spec requires:
-   can_view >= 'info' on the {prerequisite_item_id} item,
-  can_edit >= 'all' on the {dependent_item_id} item,
-  if grant_content_view = true, the user should also have can_grant_view >= 'content' on the {dependent_item_id} item,

and the user has
```json
{
	"permissions": {
		"can_view": "solution",
		"can_grant_view": "none",
		"can_watch": "answer_with_grant",
		"can_edit": "all_with_grant",
		"is_owner": true
	}
}
```
on 128139237432513103 

and 
```json
{
	"permissions": {
		"can_view": "solution",
		"can_grant_view": "none",
		"can_watch": "answer_with_grant",
		"can_edit": "all_with_grant",
		"is_owner": true
	}
}
```
on 902356740789159624.

I suspect this is due to the fact permissions returns `"can_grant_view": "none"`... but as `"is_owner": true` it should be allowed.  I don't think the spec should be changed, just the interpretation of the spec. (related with #852)

**Comment by smadbe** [comment]
On this one, the question is also either: 
- We just fix the service (probably not great)
- We fix the way permissions are queried. The `HavingMaxPermissionAtLeast` function should probably take into consideration `is_owner` in its computation.
- We consider that computed permission table should directly be computed with the max value of each permission when the permission has `is_owner=1`. But I suspect there was a reason why it wasn't done already ... ?

**Comment by GeoffreyHuck** [comment]
This should also be fixed by applying a propagation, `can_grant_view`should be max when `is_owner=1`, like this one: https://github.com/France-ioi/AlgoreaBackend/issues/852

**Comment by smadbe** [comment]
That was my third proposal 🙄

> But I suspect there was a reason why it wasn't done already ... ? If this is done, we should also make sure no service does the "is_owner" check anymore as it is useless. Actually maybe it is already done but not applied in this case (perm propagation not re-applied properly? bug?)

Also, there is no reason to have `is_owner` in `permissions_generated` as there are no propagation for it.

**Comment by GeoffreyHuck** [comment]
Haha indeed. So, I checked and there is no occurrence in the code base where a check is made on `is_owner` with another permission.

The cases where `is_owner` are used in checks are for:
- Verifying that a permission can be set:
```
func checkIfPossibleToModifyCanWatch(watchPermissionToSet string,... {
...
  if watchPermissionToSet == answerWithGrant && !managerPermissions.IsOwnerGenerated {
    return false
  }
...
}
```
- Along with `permissions_granted.can_make_session_official`:
`Where("permissions_granted.can_make_session_official OR permissions_granted.is_owner")`

That's it.

We could remove `is_owner` from `permissions_generated` but there are a few cases where the value is retrieved along with other permissions. So if we remove it, we would have to add a join to `permissions_granted` in all of those cases. All the related tests would probably have to be updated too because in many tests `permissions_generated` is defined but not `permissions_granted`. Should we do it?

**Comment by smadbe** [comment]
Probably we should plan this. For clarity, the best is probably to create a specific issue which relates to this one.

**Comment by smadbe** [comment]
Will be fixed by #966
